manuscripts2/metrics/git.py

```diff
@@ -20,50 +20,82 @@
 #
 # Author:
 #   Pranjal Aswani <aswani.pranjal@gmail.com>
-#
-
-import sys
-sys.path.insert(0, '..')
 
 from manuscripts2.elasticsearch import Query
 
 
 class GitMetrics():
+    """Root of all metric classes based on queries to a git enriched index.
+
+    This class is not intended to be instantiated, but to be
+    extened by child classes that will populate self.query with real
+    queries.
+
+    :param index: index object
+    """
+
+    def __init__(self, index, start, end):
+
+        self.query = Query(index)
+        self.start = start
+        self.end = end
+        self.query.since(self.start).until(self.end)
+
+    def timeseries(self):
+        """Obtain a time series from the current query"""
+
+        return self.query.get_timeseries(dataframe=True)
+
+    def aggregations(self):
+        """Obtain a single valued aggregation from the current query"""
+
+        return self.query.get_aggs()
+
+
+class Commits(GitMetrics):
+    """Class for computing the "commits" metric.
+
+    :param index: index object
+    """
+
+    def __init__(self, index, start, end):
+        super().__init__(index, start, end)
+        self.id = "commits"
+        self.name = "Commits"
+        self.desc = "Changes to the source code"
+        self.query = self.query.get_cardinality("hash").by_period()
+
+
+class Authors(GitMetrics):
+    """
+    class for computing the "Authors" metric
+
+    :param:
+    """
+    def __init__(self, index, start, end):
+        super().__init__(index, start, end)
+        self.id = "authors"
+        self.name = "Authors"
+        self.desc = "People authoring commits (changes to source code)"
+        self.query = self.query.get_cardinality("author_uuid").by_period()
+
+
+def overview(index, start, end):
+    """Compute metrics in the overview section for enriched git indexes.
+
+    Returns a dictionary. Each key in the dictionary is the name of
+    a metric, the value is the value of that metric. Value can be
+    a complex object (eg, a time series).
+
+    :return: dictionary with the value of the metrics
+    """
+
+    results = {
+        "activity_metrics": [Commits(index, start, end)],
```

Comment:
Why we return a list, instead of just Commits(index, start, end)?

Reply:
I was using a list for the other data sources (`github_issues`, `github_prs`) as they have more than one metric in `activity_metrics` so I just wanted to be consistent. 

I'll change it to what you suggest.

Reply:
Let's do it this way: since we have one name for each metric, let's just return the data structure for that metric, as data in a dictionary where the name of the metric is the key. If we want to group them later, for presentation, we can do it.

Reply:
Actually, @jgbarah It'll be easier for each metric (key) in the dict to have the values in a list so that we can easily iterate and add them together as one section of the report, in the report. 
It was being done previously in manuscripts also and I think it'll be easier.

Reply:
I don't understand you. For each metric, we should have only one value. If the value is complex (eg, a time series) we should have a time series. But since maybe I'm missing something, let's go this way you propose. But please, think about it, and try to find out if when you say "a list" isnt't it really, for example, several values in a data series.